pkg/protocol/sybilprotection/sybilprotectionv1/performance/snapshot.go

```diff
@@ -264,32 +265,31 @@ func (t *Tracker) exportPoolRewards(pWriter *utils.PositionedWriter, targetEpoch
 	if err := pWriter.WriteValue("pool rewards epoch count", epochCount, true); err != nil {
 		return ierrors.Wrap(err, "unable to write epoch count")
 	}
-	// TODO: make sure to adjust this loop if we add pruning later.
-	for epoch := targetEpoch; epoch > iotago.EpochIndex(0); epoch-- {
+
+	for epoch := targetEpoch; epoch > lo.Return1(t.latestPrunedEpoch()); epoch-- {
```

Comment:
This is imo not necessary and also not correct. Iiuc we want to export 365 epochs. However, this last pruned epoch might return all epochs forever in an extreme case where a node never pruned. Therefore, I'd say it's better to take the last 365 epoch (or as many as possible until and including 0) and simply have the check when we retrieve the rewards map implicit via the store itself. If for some reason an epoch is not available it'll fail (should never happen as we don't prune them for at least 365 epochs for this store).

Reply:
Amended the for loop. I hope I understood you correctly =)

Reply:
I thought about it and I'd remove lines 59 to 61 and instead, I'd put 63 to 107 inside 

`if rewardsForAccountInEpoch.PoolRewards >= rewardsForAccountInEpoch.FixedCost`

Reply:
this would zero only the epoch in question, right?

Reply:
btw, I can't find this condition in the delegator's reward function. Please see my suggestion for lines 150-177

Reply:
btw, now that I think about it, this check does no work!
See, suppose the pool reward was 15 and the fixed cost was 10. Then the function `poolReward` would spit you out 5. 
When you do this check, `rewardsForAccountInEpoch.PoolRewards<rewardsForAccountInEpoch.FixedCost=TRUE`, and the validator will be punished

Reply:
this logic at the end of funtion `poolReward` is kind of broken IMO (lines 240 to 244):

	```
if poolRewardFixedCost < fixedCost {
		return poolRewardFixedCost, nil
	}

	return poolRewardFixedCost - fixedCost, nil
```